client/src/components/Donation/PaypalButton.tsx

```diff
@@ -13,30 +14,85 @@ import envData from '../../../../config/env.json';
 import { signInLoadingSelector, userSelector } from '../../redux';
 import PayPalButtonScriptLoader from './PayPalButtonScriptLoader';
 
-const { paypalClientId, deploymentEnv } = envData;
-export class PaypalButton extends Component {
-  constructor(props) {
+type PaypalButtonProps = {
+  addDonation: (data: AddDonationData) => void;
+  donationAmount: number;
+  donationDuration: string;
+  handleProcessing: (
+    duration: string,
+    amount: number,
+    action: string
+  ) => unknown;
+  isDonating: boolean;
+  onDonationStateChange: ({
+    redirecting,
+    processing,
+    success,
+    error
+  }: {
+    redirecting: boolean;
+    processing: boolean;
+    success: boolean;
+    error: string | null;
+  }) => void;
+  skipAddDonation?: boolean;
+  t: (label: string) => string;
+  theme: string;
+  isSubscription?: boolean;
+};
+
+type PaypalButtonState = {
+  amount: number;
+  duration: string;
+  planId: string | null;
+};
+
+export interface AddDonationData {
+  redirecting: boolean;
+  processing: boolean;
+  success: boolean;
+  error: string | null;
+}
+
+const {
+  paypalClientId,
+  deploymentEnv
+}: { paypalClientId: string | null; deploymentEnv: 'staging' | 'live' } =
+  envData as {
+    paypalClientId: string | null;
+    deploymentEnv: 'staging' | 'live';
+  };
+
+export class PaypalButton extends Component<
```

Comment:
To resolve this error in the console:
```console
Warning: `PaypalButton` uses `getDerivedStateFromProps` but its initial state is undefined. This is not recommended. Instead, define the initial state by assigning an object to `this.state` in the constructor of `PaypalButton`. This ensures that `getDerivedStateFromProps` arguments have a consistent shape.
    in PaypalButton (created by withI18nextTranslation(PaypalButton))
```

I used this:
```typescript
  static displayName = 'PaypalButton';
  state: PaypalButtonState = {};
  constructor(props: PaypalButtonProps) {
    super(props);
    this.handleApproval = this.handleApproval.bind(this);
  }

  // I can't figure out if it is necessary
  // TODO: determine if this needs initializing

  static getDerivedStateFromProps(props: PaypalButtonProps): PaypalButtonState {
```

Reply:
I still get  ` Type '{}' is missing the following properties from type 'PaypalButtonState': amount, duration, planId`

how about setting it to some default values?

Reply:
Sorry, I forgot to mention the `PaypalButtonState` needs: `| Record<string, never>`